cpp/src/parquet/arrow/record_reader.cc

```diff
@@ -210,9 +212,13 @@ class RecordReader::RecordReaderImpl {
   bool nullable_values() const { return nullable_values_; }
 
   std::shared_ptr<ResizableBuffer> ReleaseValues() {
-    auto result = values_;
-    values_ = AllocateBuffer(pool_);
-    return result;
+    if (uses_values_) {
+      auto result = values_;
+      values_ = AllocateBuffer(pool_);
+      return result;
+    } else {
+      return nullptr;
+    }
```

Comment:
This is kind of a hack. I didn't want to obsess over this class since we also need to holistically look at the code duplication between RecordReader and ColumnReader, a known area of technical debt